dashboard/app/models/script_level.rb

```diff
@@ -556,4 +560,39 @@ def hidden_for_section?(section_id)
   def should_hide_survey(user, viewed_user)
     anonymous? && user.try(:teacher?) && !viewed_user.nil? && user != viewed_user
   end
+
+  def seeding_key(seed_context, use_existing_level_keys = true)
+    my_key = {
+      'script_level.level_keys': get_level_keys(seed_context, use_existing_level_keys),
+      'script_level.chapter': chapter,
+      'script_level.position': position
+    }
+
+    my_lesson = seed_context.lessons.select {|l| l.id == stage_id}.first
+    raise "No Lesson found for #{self.class}: #{my_key}, Lesson ID: #{stage_id}" unless my_lesson
+    lesson_seeding_key = my_lesson.seeding_key(seed_context)
+
+    my_key.merge!(lesson_seeding_key) {|key, _, _| raise "Duplicate key when generating seeding_key: #{key}"}
+    my_key.stringify_keys
+  end
+
+  # TODO: add validation so this is always in sync?
+  def get_level_keys(seed_context, use_existing_level_keys = true)
```

Comment:
can you please add a comment stating what this method does? I see that it gets level keys, but the way it does that seems nuanced.

Reply:
Added. It's another piece of complexity that should go away with the ScriptLevel seeding_key GUID.